backend/app/usecase/feature/dynamic_test.go

```diff
@@ -46,11 +52,66 @@ func TestDynamicDecisionMaker_IsFeatureEnable(t *testing.T) {
 				"example-feature": {
 					ID:        "example-feature",
 					IsEnabled: true,
+					Type:      entity.ManualToggle,
 				},
 			},
 			featureID:         "example-feature",
 			expectedIsEnabled: true,
 		},
+		{
+			name: "permission not found",
+			toggles: map[string]entity.Toggle{
+				"example-feature": {
+					ID:        "example-feature",
+					IsEnabled: true,
+					Type:      entity.PermissionToggle,
+				},
+			},
+			featureID: "example-feature",
+			roles: map[string][]role.Role{
+				"id": {role.Basic},
+			},
+			user: entity.User{
+				ID: "id",
+			},
+			expectedIsEnabled: false,
+		},
+		{
+			name: "permission disabled",
```

Comment:
I am curious what's this? I saw in the lines below, `IsEnabled` is true.

Reply:
User should also be admin!

Reply:
This is the condition for, "The feature is enabled, but the user is not allowed to access the feature."

Reply:
I'll change the test name to `"feature enabled, does not have permission"`